lib/matplotlib/ticker.py

```diff
@@ -1104,18 +1129,20 @@ def format_eng(self, num):
 
         mant = sign * dnum / (10 ** pow10)
 
+        # TODO: shouldn't we raise a warning if self.places < 0?
         if self.places is None:
-            format_str = "%g %s"
+            format_str = "%g{sep:s}%s".format(sep=self.sep)
         elif self.places == 0:
-            format_str = "%i %s"
+            format_str = "%d{sep:s}%s".format(sep=self.sep)
         elif self.places > 0:
-            format_str = ("%%.%if %%s" % self.places)
+            format_str = "%.{p:d}f{sep:s}%s".format(p=self.places,
+                                                    sep=self.sep)
 
         formatted = format_str % (mant, prefix)
 
         formatted = formatted.strip()
```

Comment:
I think the solution in commit 2193c79 is more or less what you were proposing. 

In fact, the `strip` at the end of `format_eng` was the last operation done before the fix #6014, which added the if statement that appends a trailing space when there was no prefix and unit (to fix issue #6009). It seems to me that with the new `space_sep` option, this `if` block is not necessary anymore and that one simply needs to perform a `strip` in `__call__` before returning the final formatted string: the only case where it will remove a trailing space should the one without a unit and without a prefix. *The relevant test seems still OK with the new scheme.*

By the way, since #6014 (and only if `space_sep=True` since #6542), `format_eng(0)` returns `u"0 "` and not `u"0"` as it is written in the docstring. So, a) is this change an issue, since `format_eng` is not a private method?, b) should I fix the docstring?